### PR TITLE
[16.0][FIX]  delivery_postlogistics: correct customer and recipient depending on picking code

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**
customer and recipient on shipping label are wrong for return picking (should be swapped in places)

### CODE REVIEW

File: delivery_postlogistics/postlogistics/web_service.py [hunk 1]
```diff
@@ -78,7 +78,12 @@ def _prepare_recipient(self, picking):
         :return a dict containing data for ns0:Recipient
 
         """
-        partner = picking.partner_id
+        if picking.picking_type_id.code == "outgoing":
+            partner = picking.partner_id
+        elif picking.picking_type_id.code == "incoming":
+            location_dest = picking.location_dest_id
+            partner = location_dest.company_id.partner_id or self.env.user.company_id
```
Comment (sebalix) [inline delivery_postlogistics/postlogistics/web_service.py]:
```suggestion
            partner = location_dest.company_id.partner_id or self.env.user.company_id.partner_id
```

But here the `partner` is not defined if the transfer is not `outgoing` or `incoming` (probably not real use cases but better to not crash if we want to use SwissPost on internal tranfers). So I would keep the original line as a default value (we often use this module to ship goods, not to receive them) and complete the code:
```python
partner = picking.partner_id
if picking.picking_type_id.code != "outgoing":
    location_dest = picking.location_dest_id
    partner = location_dest.company_id.partner_id or self.env.user.company_id.partner_id
```

File: delivery_postlogistics/postlogistics/web_service.py [hunk 2]
```diff
@@ -144,7 +149,10 @@ def _prepare_customer(self, picking):
 
         """
         company = picking.company_id
-        partner = company.partner_id
+        if picking.picking_type_id.code == "outgoing":
+            partner = company.partner_id
+        elif picking.picking_type_id.code == "incoming":
+            partner = picking.partner_id
```
Comment (sebalix) [inline delivery_postlogistics/postlogistics/web_service.py]: Same remark. Add new method to return the relevant partner to not copy the same logic.

**Review by sebalix — APPROVED** [review]

**Review by JuMiSanAr — APPROVED** [review]

**Comment by sebalix** [comment]
@ajaniszewska-dev can you remove the `[16.0]` text from the first commit message?

**Comment by ajaniszewska-dev** [comment]
> @ajaniszewska-dev can you remove the `[16.0]` text from the first commit message?

@sebalix  done

**Pull request merged**